azurelinuxagent/common/protocol/healthservice.py

```diff
@@ -0,0 +1,140 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import json
+
+from azurelinuxagent.common import logger
+from azurelinuxagent.common.exception import HttpError
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.utils import restutil
+
+
+class Observation(object):
+    def __init__(self, name, is_healthy, description='', value=''):
+        if name is None:
+            raise ValueError("Observation name must be provided")
+
+        if is_healthy is None:
+            raise ValueError("Observation health must be provided")
+
+        if value is None:
+            value = ''
+
+        if description is None:
+            description = ''
+
+        self.name = name
+        self.is_healthy = is_healthy
+        self.description = description
+        self.value = value
+
+    @property
+    def as_obj(self):
+        return {
+            "ObservationName": self.name[:64],
+            "IsHealthy": self.is_healthy,
+            "Description": self.description[:128],
+            "Value": self.value[:128]
+        }
+
+
+class HealthService(object):
+
+    ENDPOINT = 'http://{0}:80/HealthService'
+    API = 'reporttargethealth'
+    VERSION = "1.0"
+    OBSERVER_NAME = 'WALinuxAgent'
```

Comment:
>OBSERVER_NAME = 'WALinuxAgent [](start = 4, length = 29)

There should be an existing constant we can key off for this, no need for a new one.

Reply:
There is not, this is new.

Reply:
Have a constant called AGENT_NAME that has the value `WALinuxAgent`.  Why is this not the correct one to use?

Reply:
Semantically, these are distinct. We should be able to change one without changing the other.